src/csync/csync_update.cpp

```diff
@@ -387,66 +351,47 @@ static int _csync_detect_update(CSYNC *ctx, const char *file,
+      if( fs->type == CSYNC_FTW_TYPE_SLINK ) {
+          fs->error_status = CSYNC_STATUS_INDIVIDUAL_IS_SYMLINK; /* Symbolic links are ignored. */
       } else {
           if (excluded == CSYNC_FILE_EXCLUDE_LIST) {
-              st->error_status = CSYNC_STATUS_INDIVIDUAL_IGNORE_LIST; /* File listed on ignore list. */
+              fs->error_status = CSYNC_STATUS_INDIVIDUAL_IGNORE_LIST; /* File listed on ignore list. */
           } else if (excluded == CSYNC_FILE_EXCLUDE_INVALID_CHAR) {
-              st->error_status = CSYNC_STATUS_INDIVIDUAL_IS_INVALID_CHARS;  /* File contains invalid characters. */
+              fs->error_status = CSYNC_STATUS_INDIVIDUAL_IS_INVALID_CHARS;  /* File contains invalid characters. */
           } else if (excluded == CSYNC_FILE_EXCLUDE_TRAILING_SPACE) {
-              st->error_status = CSYNC_STATUS_INDIVIDUAL_TRAILING_SPACE; /* File ends with a trailing space. */
+              fs->error_status = CSYNC_STATUS_INDIVIDUAL_TRAILING_SPACE; /* File ends with a trailing space. */
           } else if (excluded == CSYNC_FILE_EXCLUDE_LONG_FILENAME) {
-              st->error_status = CSYNC_STATUS_INDIVIDUAL_EXCLUDE_LONG_FILENAME; /* File name is too long. */
+              fs->error_status = CSYNC_STATUS_INDIVIDUAL_EXCLUDE_LONG_FILENAME; /* File name is too long. */
           } else if (excluded == CSYNC_FILE_EXCLUDE_HIDDEN ) {
-              st->error_status = CSYNC_STATUS_INDIVIDUAL_EXCLUDE_HIDDEN;
+              fs->error_status = CSYNC_STATUS_INDIVIDUAL_EXCLUDE_HIDDEN;
           } else if (excluded == CSYNC_FILE_EXCLUDE_STAT_FAILED) {
-              st->error_status = CSYNC_STATUS_INDIVIDUAL_STAT_FAILED;
+              fs->error_status = CSYNC_STATUS_INDIVIDUAL_STAT_FAILED;
           } else if (excluded == CSYNC_FILE_EXCLUDE_CONFLICT) {
-              st->error_status = CSYNC_STATUS_INDIVIDUAL_IS_CONFLICT_FILE;
+              fs->error_status = CSYNC_STATUS_INDIVIDUAL_IS_CONFLICT_FILE;
           }
       }
   }
-  if (st->instruction != CSYNC_INSTRUCTION_NONE
-      && st->instruction != CSYNC_INSTRUCTION_IGNORE
-      && st->instruction != CSYNC_INSTRUCTION_UPDATE_METADATA
-      && type != CSYNC_FTW_TYPE_DIR) {
-    st->child_modified = true;
+  if (fs->instruction != CSYNC_INSTRUCTION_NONE
+      && fs->instruction != CSYNC_INSTRUCTION_IGNORE
+      && fs->instruction != CSYNC_INSTRUCTION_UPDATE_METADATA
+      && fs->type != CSYNC_FTW_TYPE_DIR) {
+    fs->child_modified = true;
   }
-  ctx->current_fs = st.get();
-
-  st->inode = fs->inode;
-  st->size  = fs->size;
-  st->modtime = fs->modtime;
-  st->type  = type;
-  st->is_hidden = fs->is_hidden;
-  st->etag = fs->etag;
-  st->file_id = fs->file_id;
-  st->directDownloadUrl = fs->directDownloadUrl;
-  st->directDownloadCookies = fs->directDownloadCookies;
-  st->remotePerm = fs->remotePerm;
-
-  // For the remote: propagate the discovered checksum
-  if (ctx->current == REMOTE_REPLICA) {
-      st->checksumHeader = fs->checksumHeader;
-  }
```

Comment:
Self-note: Is having the stat's checksum data also in the local replica going to cause issues?

Reply:
Could maybe an issue if it's outdated, we check it in `PropagateUploadFileCommon::slotComputeContentChecksum`.